lib/overcommit/hook_context/pre_commit.rb

```diff
@@ -7,7 +7,21 @@ module Overcommit::HookContext
   # This includes staged files, which lines of those files have been modified,
   # etc. It is also responsible for saving/restoring the state of the repo so
   # hooks only inspect staged changes.
-  class PreCommit < Base
+  class PreCommit < Base # rubocop:disable ClassLength
+    # Returns whether this hook run was triggered by `git commit --amend`
+    def amend?
+      if @amended.nil?
+        cmd = Overcommit::Utils.parent_command
+        @amended = !(/--amend/ =~ cmd).nil?
+
+        amend_alias = `git config --get-regexp '^alias\\.' '--amend'`.
+          slice(/(?<=alias\.)\w+/)
+
+        @amended ||= !(/git #{amend_alias}/ =~ cmd).nil? unless amend_alias.nil?
```

Comment:
I don't believe the `unless amend_alias.nil?` here is necessary, since if it's `nil`/`false` the `||` would have resulted in the same resulting outcome. If you implement my early-return suggestion above, you don't even need `||=`.

On another note, I think it make make this code slightly more legible to first-time viewers if we had a quick comment describing what each of these mysterious blocks of regexes are doing.